packages/template-retail-react-app/app/utils/product-utils.js

```diff
@@ -33,3 +33,21 @@ export const getDisplayVariationValues = (variationAttributes, values = {}) => {
     }, {})
     return returnVal
 }
+
+/**
+ * This function extract the promotional price from a product. If there are more than one price, the smallest price will be picked
+ * @param {object} product - product detail object
+ * @returns {{discountPrice: (number|number), basePrice: *}}
+ */
+export const getDisplayPrice = (product) => {
+    const basePrice = product?.pricePerUnit || product?.price
+    const promotionalPriceList = product?.productPromotions
+        ?.map((promo) => promo.promotionalPrice)
+        .filter(Boolean)
+    // choose the smallest price among the promotionalPrice
+    const discountPrice = promotionalPriceList?.length ? Math.min(...promotionalPriceList) : 0
```

Comment:
```suggestion
    const discountPrice = promotionalPriceList?.length ? Math.min(...promotionalPriceList) : null
```
❗ We need some way to distinguish "available for free" from "no discounted price".